org.adempiere.base/src/org/compiere/model/PO_Record.java

```diff
@@ -59,46 +60,63 @@
 	/**
 	 * 	Delete Cascade including (selected)parent relationships
 	 *	@param AD_Table_ID table
-	 *	@param Record_ID record
+	 *	@param Record_IDorUU record ID (int) or UUID (String)
 	 *	@param trxName transaction
 	 *	@return false if could not be deleted
 	 */
-	static boolean deleteRecordIdCascade (int AD_Table_ID, int Record_ID, String trxName)
+	static boolean deleteRecordCascade (int AD_Table_ID, Serializable Record_IDorUU, String trxName)
 	{
-		KeyNamePair[] cascades = getTablesWithRecordIDColumn(MColumn.FKCONSTRAINTTYPE_ModelCascade, trxName);
+		int refId;
+		String columnName;
+		if (Record_IDorUU instanceof Integer) {
+			refId = DisplayType.RecordID;
+			columnName = "Record_ID";
+		} else if (Record_IDorUU instanceof String) {
+			refId = DisplayType.RecordUU;
+			columnName = "Record_UU";
+		} else {
+			log.warning(Record_IDorUU.getClass().getName() + " not supported for ID/UUID");
+			return true;

```

Comment:
better throw illegalargumentexception here. should never happens.

Reply:
done